### ISSUE TITLE
different accuracy on lfw between training and testing

### ISSUE THREAD

**Issue by qunluo** [deepinsight/insightface#85]
Hi, I use ms1m dataset to train r100 model.
During training, the accuracy on lfw is 0.998 as follow:
testing verification..
(12000, 512)
infer time 62.720226
[lfw][308000]XNorm: 22.075580
[lfw][308000]Accuracy-Flip: 0.99800+-0.00245
testing verification..
(14000, 512)
infer time 76.100626
[cfp_ff][308000]XNorm: 20.756458
[cfp_ff][308000]Accuracy-Flip: 0.99800+-0.00214
testing verification..
(14000, 512)
infer time 67.842646
[cfp_fp][308000]XNorm: 20.607653
[cfp_fp][308000]Accuracy-Flip: 0.93457+-0.01542
testing verification..
(12000, 512)
infer time 68.593957
[agedb_30][308000]XNorm: 22.874611
[agedb_30][308000]Accuracy-Flip: 0.97700+-0.00756
saving 77
INFO:root:Saved checkpoint to "../model-r100-0077.params"

After saving the model-r100-0077.params, I use do_ver.sh to test the model on lfw, but only get 0.99733 on lfw:
python -u verification.py --gpu 1 --data-dir ../../datasets/faces_ms1m_112x112  --model '../../model-r100-softmax1e4/model-r100,77' --target lfw --batch-size 64
image_size [112, 112]
model number 1
loading ../../model-r100-softmax1e4/model-r100 77
[16:38:31] src/nnvm/legacy_json_util.cc:204: Warning: loading symbol saved by MXNet version 10100 with lower version of MXNet v10000. May cause undefined behavior. Please update MXNet if you encounter any issue
[16:38:41] src/operator/././cudnn_algoreg-inl.h:107: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
model loading time 13.996371
loading..  lfw
loading bin 1000
loading bin 2000
loading bin 3000
loading bin 4000
loading bin 5000
loading bin 6000
loading bin 7000
loading bin 8000
loading bin 9000
loading bin 10000
loading bin 11000
loading bin 12000
(12000L, 3L, 112L, 112L)
testing verification..
(12000, 512)
infer time 207.145642
[lfw]XNorm: 22.055374
[lfw]Accuracy: 0.00000+-0.00000
[lfw]Accuracy-Flip: 0.99733+-0.00291
Max of [lfw] is 0.99733

**Comment by nttstar** [comment]
I also found it recently but I don't know why. Hope somebody can help us.

**Comment by aitaoquanyi** [comment]
@qunluo maybe the XNorm: 22.055374/XNorm: 22.075580?

**Comment by myfighterforever** [comment]
I have test my model, it is the same result.

**Comment by myfighterforever** [comment]
@qunluo This is my log:
image_size [112, 112]
model number 1
loading ../../model-r100 85
[06:14:03] src/operator/././cudnn_algoreg-inl.h:107: Running performance tests to find the best convolution algorithm, this can take a while... (setting env variable MXNET_CUDNN_AUTOTUNE_DEFAULT to 0 to disable)
model loading time 11.679833
loading..  lfw
loading bin 1000
loading bin 2000
loading bin 3000
loading bin 4000
loading bin 5000
loading bin 6000
loading bin 7000
loading bin 8000
loading bin 9000
loading bin 10000
loading bin 11000
loading bin 12000
(12000L, 3L, 112L, 112L)
testing verification..
(12000, 512)
infer time 70.645818

I think your problem maybe caused by this" loading symbol saved by MXNet version 10100 with lower version of MXNet v10000. May cause undefined behavior. Please update MXNet if you encounter any issue"

**Comment by nttstar** [comment]
#111

**Comment by qunluo** [comment]
I use the same machine for training and testing, but I still get different results on lfw. Like 99.817% on training and 99.8% on testing(do_ver.sh).
